shared/src/store/reducers/user-configuration.ts

```diff
@@ -310,22 +310,45 @@ function generateMacroId(macros: Macro[]) {
     return newId + 1;
 }
 
-function checkExistence(layers: Layer[], property: string, value: any) {
-    const newLayers = layers.map(layer => {
-        const newLayer = new Layer(layer);
-
-        newLayer.modules = layer.modules.map((module: Module) => {
-            module.keyActions.forEach((action: KeyAction, index: number) => {
+function checkExistence(layers: Layer[], property: string, value: any): Layer[] {
+    const keyActionsToClear: {
+        layer: number,
+        module: number,
+        keyAction: number
+    }[] = [];
+    for (let i = 0; i < layers.length; ++i) {
```

Comment:
Then possibly `layerIdx`, `moduleIdx`, and `keyActionIdx` would be better names. Normally, I support simple cycle variable names but in this case there are 3 of them and I really feel strongly that this would improve readability.